src/error.rs

```diff
@@ -29,6 +29,10 @@ where
 
     /// The server sent a response that the client was not expecting.
     Desynchronized,
+
+    /// The underlying transport task did not exit gracefully (either panic or cancellation).
+    /// Transport task panics can happen for example when the codec logic panics.
+    TransportUnexpectedExit,
```

Comment:
Nit: maybe call this `TransportDropped` since that tracks closer to what the actually issue is?

Reply:
Sounds good, done